assets/js/components/legacy-notifications/util.js

```diff
@@ -106,41 +113,36 @@ const removeDismissed = async ( notifications ) => {
  * Gets notifications from session storage, fallback to notifications API request.
  *
  * @since 1.0.0
+ * @since n.e.x.t Memoized to prevent duplicate simultaneous fetch requests from different callers.
  *
- * @return {number} Number of module notifications.
+ * @return {Promise} Object with `results` (map [slug]: notificationObject[]) and `total` (int).
  */
-export async function getModulesNotifications() {
+export const getModulesNotifications = memize( async () => {
```

Comment:
Memoizing the function works as expected here because it's essentially memoizing a promise immediately rather than the final resolved value of the function.